packages/client/src/views/UserSetup/SetupReviewPage.tsx

```diff
@@ -72,137 +65,139 @@ const LoaderOverlay = styled.div`
         action: {
+          id: `Question_Action_${e.questionKey}`,
           label: intl.formatMessage(buttonMessages.change),
-          disabled: true
+          handler: () =>
+            goToStep(ProtectedAccoutStep.SECURITY_QUESTION, setupData)
         }
-      },
-      {
-        id: 'RegisterOffice',
-        label: intl.formatMessage(messages.labelAssignedOffice),
-        value: primaryOffice
-      },
-      {
-        id: 'RoleType',
-        label: `${intl.formatMessage(
-          constantsMessages.labelRole
-        )} / ${intl.formatMessage(constantsMessages.type)}`,
-        value: typeRole
-      },
-      ...answeredQuestions
-    ]
-
-    const onCompleted = () => {
-      this.props.goToStep(
-        ProtectedAccoutStep.CONFIRMATION,
-        this.props.setupData
-      )
-    }
-    const onError = () => {
-      this.setState({
-        submitError: true
       })
-    }
-    return (
-      <ActionPageLight
-        title={intl.formatMessage(messages.userSetupRevieTitle)}
-        goBack={() => {
-          this.props.goToStep(
-            ProtectedAccoutStep.SECURITY_QUESTION,
-            this.props.setupData
+    })
+  const items = [
+    {
+      id: 'EnglishName',
+      label: intl.formatMessage(messages.labelEnglishName),
+      value: englishName,
+      action: {
+        label: intl.formatMessage(buttonMessages.change),
+        disabled: true
+      }
+    },
+    {
+      id: 'UserPhone',
+      label: intl.formatMessage(constantsMessages.labelPhone),
+      value: mobile,
+      action: {
+        label: intl.formatMessage(buttonMessages.change),
+        disabled: true
+      }
+    },
+    {
+      id: 'RegisterOffice',
+      label: intl.formatMessage(messages.labelAssignedOffice),
+      value: primaryOffice
+    },
+    {
+      id: 'RoleType',
+      label: `${intl.formatMessage(
+        constantsMessages.labelRole
+      )} / ${intl.formatMessage(constantsMessages.type)}`,
+      value: typeRole
+    },
+    ...answeredQuestions
+  ]
+
+  const onCompleted = () => {
+    goToStep(ProtectedAccoutStep.CONFIRMATION, setupData)
+  }
+  const onError = () => {
+    setSubmitError(true)
+  }
+  const confirmActionButton = (
+    <Mutation
+      mutation={activateUserMutation}
+      variables={{ ...setupData }}
+      onCompleted={() => onCompleted()}
+      onError={() => onError()}
+    >
+      {(submitActivateUser: any, { loading }: { loading: any }) => {
+        if (loading) {
+          return (
+            <LoaderOverlay>
+              <Loader
+                id="setup_submit_waiting"
+                loadingText={intl.formatMessage(messages.waiting)}
+              />
+            </LoaderOverlay>
           )
-        }}
+        }
+        return (
+          <ConfirmButton id="Confirm" onClick={() => submitActivateUser()}>
+            <Check />
+            {intl.formatMessage(buttonMessages.confirm)}
+          </ConfirmButton>
+        )
+      }}
+    </Mutation>
+  )
```

Comment:
We can probably extract this into a component and could we type the mutation properly perhaps?